homeassistant/components/flo/device.py

```diff
@@ -0,0 +1,176 @@
+
+from .const import DOMAIN as FLO_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class FloDeviceDataUpdateCoordinator(DataUpdateCoordinator):
+    """Flo device object."""
+
+    def __init__(
+        self, hass: HomeAssistantType, api_client: API, location_id: str, device_id: str
+    ):
+        """Initialize the device."""
+        self.hass: HomeAssistantType = hass
+        self.api_client: API = api_client
+        self._flo_location_id: str = location_id
+        self._flo_device_id: str = device_id
+        self._manufacturer: str = "Flo by Moen"
+
+        self._mac_address: str = None
+        self._model: str = None
+        self._device_type: str = None
+        self._serial_number: str = None
+        self._available: bool = None
+        self._firmware_version: str = None
+        self._rssi: int = None
+        self._last_heard_from_time: str = None
+        self._current_system_mode: str = None
+        self._target_system_mode: str = None
+        self._current_flow_rate: float = None
+        self._current_psi: float = None
+        self._temperature: float = None
+        self._consumption_today: float = None
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{FLO_DOMAIN}-{device_id}",
+            update_interval=timedelta(seconds=60),
+        )
+
+    async def _async_update_data(self):
+        """Update data via library."""
+        try:
+            async with timeout(10):
+                await asyncio.gather(
+                    *[self._update_device(), self._update_consumption_data()]
+                )
+        except (RequestError) as error:
+            raise UpdateFailed(error)
+
+    @property
+    def location_id(self) -> str:
+        """Return Flo location id."""
+        return self._flo_location_id
+
+    @property
+    def id(self) -> str:
+        """Return Flo device id."""
+        return self._flo_device_id
+
+    @property
+    def device_name(self) -> str:
+        """Return device name."""
+        return f"{self.manufacturer} {self.model}"
+
+    @property
+    def mac_address(self) -> str:
+        """Return ieee address for device."""
+        return self._mac_address
+
+    @property
+    def manufacturer(self) -> str:
+        """Return manufacturer for device."""
+        return self._manufacturer
+
+    @property
+    def model(self) -> str:
+        """Return model for device."""
+        return self._model
+
+    @property
+    def rssi(self) -> float:
+        """Return rssi for device."""
+        return self._rssi
+
+    @property
+    def last_heard_from_time(self) -> str:
+        """Return lastHeardFromTime for device."""
+        return self._last_heard_from_time
+
+    @property
+    def device_type(self) -> str:
+        """Return the device type for the device."""
+        return self._device_type
+
+    @property
+    def available(self) -> bool:
+        """Return True if device is available."""
+        return self._available
```

Comment:
This should include if last update was a success.

Reply:
updated